apm-agent-core/src/main/java/co/elastic/apm/impl/transaction/Id.java

```diff
@@ -62,15 +83,15 @@ public void resetState() {
         cachedStringRepresentation = null;
```

Comment:
The problem is you could do:

```java
Id id = Id.new128BitId();
id.setToRandomValue();
String idAsString = id.toString();
id.setToRandomValue();
assert idAsString != id.toString();
```

which would make the assertion fail because the string representation of the first `setToRandomValue` is still cached.